[ReutersBridge] Add New Bridge

hollowleviathan opened the pull request:
Adds bridge requested in #1618

hollowleviathan committed:
[Reuters] New bridge

Spraynard committed:
Modifying ReutersBridge

News Feed types now selectable through a list parameter

JSON return is now being parsed correctly

Stories being gathered in a stepwise process

hollowleviathan committed:
[Reuters] Fix code style

csisoap commented:
Bridge is working perfectly.
Thank you.
But it seem "Pictures" section isn't working. I got blank page on that section.

Query string: `action=display&bridge=Reuters&feed=pictures&format=Html`

somini commented:
See https://github.com/RSS-Bridge/rss-bridge/pull/1651#issuecomment-658454626

csisoap commented:
![image](https://user-images.githubusercontent.com/33269526/87751391-15e81600-c828-11ea-9845-30b16369a5e1.png)

It would be great if it can get the article from "Editor's Highlight" in the above picture. It seem they are different than the "story" type.

csisoap committed:
[ReutersBridge] Add Editor's Highlight to feed, add feed name, and add more categories

hollowleviathan committed:
[Reuters] Fix code style

csisoap committed:
Modifying ReutersBridge (#5)

[ReutersBridge] Add all article from 'Editor\'s Highlight' to the feed, more categories, author name, full article text.

Spraynard commented:
Are there any issues regarding merging this PR? I can help get everything ready for merging if the project needs it

hollowleviathan commented:
The primary concern from the RSS-Bridge team seems to be checking if https://github.com/RSS-Bridge/rss-bridge/pull/1651 does anything this PR doesn't and incorporating it. Since that PR is a scrape of the UK reuters site, I'm daunted by figuring out how to understand what advantages, if any, it has over using the direct API.

I also notice there's a bug where this PR has some tag text gunk at the end of articles "for-phone-onlyfor-tablet-portrait-upfor-tablet-landscape-upfor-dekstop-upfor-wide-desktop-up"

Spraynard commented:
Alright thank you for giving me an update! I'll see if there's anything that can be added

csisoap commented:
> The primary concern from the RSS-Bridge team seems to be checking if #1651 does anything this PR doesn't and incorporating it. Since that PR is a scrape of the UK reuters site, I'm daunted by figuring out how to understand what advantages, if any, it has over using the direct API.
> 
> I also notice there's a bug where this PR has some tag text gunk at the end of articles "for-phone-onlyfor-tablet-portrait-upfor-tablet-landscape-upfor-dekstop-upfor-wide-desktop-up"

Can i ask where you get that bug from? 

I think the advantage of this using direct API is that you get all info of that article (id, timestamp, article's content,...).

hollowleviathan commented:
> Can i ask where you get that bug from?

Observation from using this PR myself.

> I think the advantage of this using direct API is that you get all info of that article (id, timestamp, article's content,...).

Yes, I think using the direct API is superior and *should* get all of the news items that a scrape gets and more, but how do you prove that?

Spraynard commented:
> Yes, I think using the direct API is superior and should get all of the news items that a scrape gets and more, but how do you prove that?

From what I understand I don't think people are arguing to keep the scraping method, I just think people are wondering if there's any more information obtained from the scrape, (or even a different source of information) than what is being obtained in this bridge.

~~Just briefly looking while writing this comment, our bridge is pulling from `https://reuters.com` whereas #1651 bridge data is coming from `https://uk.reuters.com`. We might have to put a selector in our bridge to merge in `uk.reuters` data.~~ Sorry I'm trying to remember what is actually happening and I was ignorant

Spraynard commented:
[This linked pull request](https://github.com/hollowleviathan/rss-bridge/pull/10) to the `reuters` branch of @hollowleviathan's repo should combine the functionality between this PR and #1651. 

Selecting US will present stories and link back to `reuters.com`, whereas selecting UK will link back to `uk.reuters.com`.

Spraynard committed:
Adding region selection to ReutersBridge (#10)

Adding region selection to ReutersBridge

csisoap commented:
The PR is now ready to be merged. It's just need to get merge with PR #1651.

Spraynard commented:
@csisoap Sorry I am unsure what you mean by _merge_ with the other PR. I looked at what #1651 was doing and saw that it was looking at stories from `uk.reuters.com` and so I added the region contexts. I think that we can merge this and have to ask #1651 to close their pull request 😔

csisoap commented:
> @csisoap Sorry I am unsure what you mean by _merge_ with the other PR. I looked at what #1651 was doing and saw that it was looking at stories from `uk.reuters.com` and so I added the region contexts. I think that we can merge this and have to ask #1651 to close their pull request 😔

Yeah, my fault. Sorry about that, i'm terrible at English :(

About asking #1651 , i tried to ask him to check this PR but i haven't received any response.

Spraynard commented:
You're fine :) English is a dumb language anyway.

Hmmm okay. Well I don't know who reviews these pull requests but since this one seems to be more robust and is working fine I think it can be merged. I don't know if #1651 is viable anymore

csisoap commented:
@Spraynard it seem there only @em92 doing review for PRs right now.

em92 commented on bridges/ReutersBridge.php:
```diff
@@ -0,0 +1,277 @@
+					'Sports' => 'sports',
+					'USA News' => 'us',
+				)
+			)
```
I suggest to extract common catogories to some array. Then use array_merge to add non-common ones.

csisoap replied:
Hi, i tried to split up common categories to some array and when i tried to merge, the compiler doesn't happy with me as const don't allow expression. Same happen with private. I don't know how to merge two array into one giant parameter as you saw in the code. There are spread operator which can use to merge but it's on PHP 7.4 which may failed some test for older version.

I just hope this bridge will soon to be merged.

em92 reviewed (commented):
UK and US feeds look equal.
https://feed.eugenemolotov.ru/pr1653/?action=display&bridge=Reuters&context=United+States&feed_us=tech&format=Html
https://feed.eugenemolotov.ru/pr1653/?action=display&bridge=Reuters&context=United+States&feed_uk=tech&format=Html

Is that expected?

csisoap commented:
> UK and US feeds look equal.
> https://feed.eugenemolotov.ru/pr1653/?action=display&bridge=Reuters&context=United+States&feed_us=tech&format=Html
> https://feed.eugenemolotov.ru/pr1653/?action=display&bridge=Reuters&context=United+States&feed_uk=tech&format=Html
> 
> Is that expected?

You have to change the context param to United Kingdom on the second link to see what different between two feeds. Sometimes they equal each other, but if you look into other feed like World, Top News, you'll see the order between is different.

csisoap committed:
[ReutersBridge] New change (#11)

- Health feed in the U.S context now using different feed.
- Bridge now support using a unique feed ID instead of normal name.
- Remove U.K context.
- Add some new feeds.

em92 commented:
Alphabetical order is required. Hard to find category, that user could be interested in.

![Снимок экрана_2021-01-22_10-01-03](https://user-images.githubusercontent.com/15870400/105448909-0f37d100-5c99-11eb-899f-d33b36734fbd.png)

@Spraynard and @hollowleviathan. There was a discussion about UK and US versions of reuters. Is it fine, that latest changes remove separated US and UK feeds?

csisoap commented:
Hi @em92 
The reason to remove U.K context was that Reuters merged several edition, includes U.K, into one global site so i personally think it would not necessary to use U.K context anymore.
I have added all feeds that specfic to U.K edition into U.S context.

You can find their statement [here](https://www.reutersagency.com/en/coverage/changes-to-your-reuters-regional-experience-uk-in/).

Spraynard committed:
Updating ReutersBridge

Sorting selection  of news feed values

Spraynard commented:
News feed items have been sorted with [this](https://github.com/hollowleviathan/rss-bridge/pull/12) pull request. Once we get that merged we should fit the _sorted feed items_ criteria.

em92 commented:
> You can find their statement [here](https://www.reutersagency.com/en/coverage/changes-to-your-reuters-regional-experience-uk-in/).

Fair enough.



> News feed items have been sorted with [this](https://github.com/hollowleviathan/rss-bridge/pull/12) pull request. Once we get that merged we should fit the _sorted feed items_ criteria.

@hollowleviathan waiting for you to push mentioned PR.

csisoap commented:
Pinging @hollowleviathan just in case he forgot.

hollowleviathan committed:
Merge pull request #12 from Spraynard/reuters

Updating ReutersBridge

hollowleviathan commented:
Sorry about the delay, merged now.

em92 commented:
gj you all!

Spraynard commented:
Amazing!!!!!
